tests/app/questionnaire/test_placeholder_parser.py

```diff
@@ -1137,3 +1137,99 @@ def test_placeholder_parser_calculated_summary_dependencies_cache(
     placeholder_2 = placeholder_parser(placeholder_list=placeholder_list_2)
     assert placeholder_2["unit-total-playback"] == 11
     assert path_finder.called == 1
+
+
+def test_placeholder_dependencies_cache(mocker, mock_renderer):
+    schema = load_schema_from_name("test_first_non_empty_item")
+    path_finder = mocker.patch("app.questionnaire.path_finder.PathFinder.routing_path")
```

Comment:
I think it is easy enough to initialise path finder here and just mock, or make up some of its inputs, like we do elsewhere:
```python
schema = load_schema_from_name("test_skip_condition_group")
    answer_store.add_or_update(
        Answer(answer_id="do-you-want-to-skip-answer", value="Yes")
    )

    # When
    path_finder = PathFinder(schema, answer_store, list_store, progress_store, None, {})
```
remember schema and answer store, list store can be only minimal objects.

Reply:
Added 👍  The path finder check is also different now